added `Wikipedia` retriever

leo-gan opened the pull request:
- added `Wikipedia` retriever. It is effectively a wrapper for `WikipediaAPIWrapper`. It wrapps load() into get_relevant_documents()
- sorted `__all__` in the `retrievers/__init__`
- added integration tests for the WikipediaRetriever
- added an example (as Jupyter notebook) for the WikipediaRetriever

leo-gan committed:
created retrievers/wikipedia.py and integration tests

leo-gan committed:
created wikipedia.ipynb; added class to the retrievers.__init__

leo-gan committed:
fixed lint-s

leo-gan committed:
added wikipedia package to [tool.poetry.group.test_integration.dependencies]

leo-gan committed:
moved tests from unit tests to integration tests

hwchase17 reviewed (commented):
will this work out of the box in a question answer chain? can you show an example of what that would look like?

leo-gan committed:
added limit on the document.content size; add a question answering chain example

leo-gan commented:
> 

@hwchase17 I've added an example with ConversationalRetrievalChain to the notebook. I also added a default limit size to the Document content. 
Note: I've commented out a line in the example
`# "How big is Wikipédia en français?",` because if fails. I'm not sure it fails in my code. Could you, please, check it?

dev2049 commented on langchain/utilities/wikipedia.py:
```diff
@@ -104,12 +106,12 @@ def load(self, query: str) -> List[Document]:
         page_titles = self.wiki_client.search(query[:WIKIPEDIA_MAX_QUERY_LENGTH])
         docs = []
-        for page_title in page_titles[: self.top_k_results]:
+        for page_title in page_titles[: self.doc_content_chars_max]:
```
is page_titles a string or a list of things? was this line supposed to be changed?

leo-gan replied:
A good catch! Thanks! Yes, it should be `top_k_results`.

leo-gan committed:
fixed a bug

dev2049 commented:
this is awesome, thanks @leo-gan!
